subsys/bluetooth/audio/stream.c

```diff
@@ -27,12 +27,43 @@
 #define LOG_MODULE_NAME bt_audio_stream
 #include "common/log.h"
 
+static uint8_t pack_bt_codec_cc(const struct bt_codec *codec, uint8_t cc[])
+{
+	uint8_t len;
+
+	len = 0U;
+	for (size_t i = 0U; i < codec->data_count; i++) {
+		const struct bt_data *data = &codec->data[i].data;
+
+		/* We assume that data_len and data has previously been verified
+		 * and that based on the Kconfigs we can assume that the length
+		 * will always fit in `cc`
+		 */
+		(void)memcpy(cc + len, data->data, data->data_len);
+		len += data->data_len;
+	}
+
+	return len;
+}
+
+void bt_audio_codec_to_iso_path(struct bt_iso_chan_path *path,
+				const struct bt_codec *codec)
+{
+	BT_ERR("bt_audio_codec_to_iso_path");
+	path->pid = codec->path_id;
+	path->format = codec->id;
+	path->cid = codec->cid;
+	path->vid = codec->vid;
+	path->delay = 0; /* TODO: Add to bt_codec? Use presentation delay? */
```

Comment:
I believe this is _part_ of the presentation delay.

Reply:
Hmm, maybe it's related. But it's described as:
```c
When Data_Path_Direction is set to 0x00 (input), the Controller_Delay param-
eter specifies the delay at the data source from the reference time of an SDU to
the CIG reference point (see [Vol 6] Part B, Section 4.5.14.1) or BIG anchor
point (see [Vol 6] Part B, Section 4.4.6.4). When Data_Path_Direction is set to
0x01 (output), Controller_Delay specifies the delay from the
SDU_Synchronization_Reference to the point in time at which the Controller
begins to transfer the corresponding data to the data path interface. The Host
should use the HCI_Read_Local_Supported_Controller_Delay command to
obtain a suitable value for Controller_Delay.
```

So it is a delay between the local controller and the host, and not the delay between the two hosts.

Reply:
I think that fits well with being a part of the presentation delay.

From BAP 7.1.1: "For reception of a frame of audio data by a Unicast Server in the Audio Sink role or by a Broadcast Sink, the Presentation_Delay parameter value is the time period between the SDU Synchronization Reference and the start of presentation of that frame of audio data"
And from BAP 7.1: "The value of the Presentation_Delay parameter includes any implementation-specific delays in the Unicast Server and/or Broadcast Sink, such as processing time for internal transports, codec processing, ADC/DAC delays, application-specific audio processing, etc."

The delay above gives the time from the SDU reference point until data are transferred to the host.  The host will then spend some time decoding, and the remaining part of the presentation delay, the host/application must buffer the data.

Reply:
For incoming data it can perhaps be used, although it won't be able to take any processing at the host or application layer into account (including decoding). 

So We cannot just set this to the presentation delay if we are doing ISO over HCI.

Reply:
I think that is just my point.
But the comment seems to indicate that we might do that (set it to presentation delay).
It's a comment, with a question mark, though.